### ISSUE TITLE
Indent rule seems wrong for returning a large array/object

### ISSUE THREAD

**Issue by jpanderson-outreach** [eslint/eslint#8014]
**Tell us about your environment**

* **ESLint Version:** `v3.14.1`
* **Node Version:** `v6.9.2`
* **npm Version:** `3.10.8`

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
```
module.exports = {
  "parser": "babel-eslint", // Allows us to use all the awesome es6/es7 features that aren't yet supported natively in eslint
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "rules": {
    "curly": 2,
    "indent": ["error", 2, {"SwitchCase": 1}],
    "no-debugger": 2,
    "no-dupe-keys": 2,
    "no-trailing-spaces": 2,
    "react/jsx-no-duplicate-props": 2,
    "react/jsx-no-undef": 2,
    "react/no-deprecated": 2,
    "react/no-direct-mutation-state": 2,
    "react/no-find-dom-node": 2,
    "react/react-in-jsx-scope": 2,
    "react/require-render-return": 2,
    "react/require-render-return": 2,
    "semi": 2,
  },
  "fix": true,
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "plugins": ['react']
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
const foobar = () => {
  return (
    [
      1,
      2,
      3,
      4
    ]
  );
};

const foobar = () => {
  return (
    {
      a: 'apple',
      b: 'bear'
    }
  );
};
```

**What did you expect to happen?**
Would expect no warnings

**What actually happened? Please include the actual, raw output from ESLint.**
```
  3:5  error  Expected indentation of 2 spaces but found 4  indent
  4:7  error  Expected indentation of 4 spaces but found 6  indent
  5:7  error  Expected indentation of 4 spaces but found 6  indent
  6:7  error  Expected indentation of 4 spaces but found 6  indent
  7:7  error  Expected indentation of 4 spaces but found 6  indent
  8:5  error  Expected indentation of 2 spaces but found 4  indent

  14:5  error  Expected indentation of 2 spaces but found 4  indent
  15:7  error  Expected indentation of 4 spaces but found 6  indent
  16:7  error  Expected indentation of 4 spaces but found 6  indent
  17:7  error  Expected indentation of 4 spaces but found 6  indent
```

The following code achieves no errors (obviously):
```
const foobar = () => {
  return (
  [
    1,
    2,
    3,
    4
  ]
  );
};

const foobar = () => {
  return (
  {
    a: 'apple',
    b: 'bear'
  }
  );
};
```

This may be intended, but it certainly looks...strange, at least to my eyes.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, I'm going to close this as a duplicate of https://github.com/eslint/eslint/issues/6007.

This is fixed by https://github.com/eslint/eslint/pull/7618.
